### PR TITLE
Encode Speed in URL

### PR DESCRIPTION
**Pull request by unfokus**
Let's spam into the browser history of all the people \o/

Is this how to javascript? uwu

### CODE REVIEW

**Comment by unfokus** [comment]
Proposal for less spam:
Only push the URL into the History at specific actions:
* Textbox: onfocusout()
* Slider: onmouseup()

File: src/App.js [hunk 1]
```diff
@@ -65,7 +67,7 @@ class Machine extends Component {
 
     handleChange(event) {
         this.setState({source: event.target.value});
-        history.replaceState({source: event.target.value}, 'source', '?s=' + event.target.value);
+        history.replaceState({source: this.state.source, speed: this.state.speed}, 'Colorfuck', '?s=' + this.state.source + "&speed=" + this.state.speed );
```
Comment (leftshift) [inline src/App.js]: `setState()` is an asynchronous  operation, so accessing `this.state` immediately afterwards might still give the old value. We'll either have to use `event.target.value` or move the `history.replace state(…)` to a callback function passed to `setState` as a second argument

File: src/App.js [hunk 2]
```diff
@@ -82,8 +84,10 @@ class Machine extends Component {
                 }
             }
         )
+        history.replaceState({source: this.state.source, speed: this.state.speed}, 'Colorfuck', '?s=' + this.state.source + "&speed=" + this.state.speed );
     }
```
Comment (leftshift) [inline src/App.js:95]: Same as above. It's probably easiest just to move it into the callback function above.

**Comment by leftshift** [comment]
Also, not spamming history too much is a good idea. I think we're only actually pushing the source on starting and just replacing it on change. For the speed, let's maybe just replace and never push at all, so going back in history goes back to the previous source with it's last set speed

**Pull request merged**